python/src/scipp/plotting/resampling_model.py

```diff
@@ -180,16 +181,20 @@ def __init__(self, array, **kwargs):
         super().__init__(self._to_density(array), **kwargs)
 
     def _to_density(self, array):
-        array = PlotArray(data=array.data.astype(sc.dtype.float64),
-                          meta=array.meta,
-                          masks=array.masks)
-        for dim in array.data.dims:
-            coord = array.meta[dim]
+        if array.dtype != sc.dtype.float64:
+            data = array.data.astype(sc.dtype.float64)
+        else:
+            data = array.data
+        density = sc.DataArray(data=data,
+                               coords=array.coords.to_dict(),
+                               masks=array.masks.to_dict())
```

Comment:
```suggestion
        array = sc.astype(array, sc.dtype.float64)
```
and keep calling it `array` below.

Reply:
hmm yes, I had started with that but then remembered that `astype` always makes a new object, even if the dtype is the same.
But you are implying this is ok because even if `astype` is making a copy, it is cheap because of the shared ownership?

Reply:
Yes, it is copying the object but not the buffers.